appender_test.go

```diff
@@ -117,6 +133,45 @@ loop:
 		AvailableBulkRequests: 10,
 		BytesTotal:            bytesTotal,
 	}, stats)
+
+	var rm metricdata.ResourceMetrics
+	assert.NoError(t, rdr.Collect(context.Background(), &rm))
+	var asserted int
+	assertCounter := func(metric metricdata.Metrics, count int64, attrs attribute.Set) {
+		asserted++
+		// assert.Equal(t, "s", metric.Unit)
+		counter := metric.Data.(metricdata.Sum[int64])
+		for _, dp := range counter.DataPoints {
+			assert.Equal(t, count, dp.Value)
+			assert.Equal(t, attrs, dp.Attributes)
+		}
+	}
+	wantAttrs := attribute.NewSet(indexerAttrs...)
+	for _, metric := range rm.ScopeMetrics[0].Metrics {
```

Comment:
What this doesn't tell us is that all of the metrics are reported - only _if_ they're recorded, that their values are correct.

Should we use the https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest?

Reply:
I think the purpose of this test is to assert that the values are recorded correctly, but not necessarily reported by the exporter, what do you think? 

Also, I had a look at the `[metricdatatest](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest) package, and it seems it is a bit more complicated to use that what we already have here.

Reply:
I missed the "asserted" check. It looks good now with the `default` case.
Thanks for looking into `metricdatatest`, sounds fine.